### PR TITLE
Feat/309 search guides stories and actions

### PR DESCRIPTION
**Pull request by katjam**
Partial #309

## Description

- Lint
- WIP search results wired up to view.
- Bug: Encoded search string is empty so always no results

@geeksforsocialchange/developers

### CODE REVIEW

**Comment by katjam** [comment]
If I can't get this working for the morning - we should remove the search box before people start testing.

@r-ferrier can you check where it's at in the morning and liaise with @AutumnBarlow? I think the best solution is to take out the search box and let people test without it until it's working. That will cause the least confusion. I'll get it fixed over the weekend either way but I can't guarantee internet tomorrow.

File: src/Theme/HeaderTemplate.elm [hunk 1]
```diff
@@ -65,36 +63,32 @@ searchInput model =
...
+                        |> Page.Search.Data.searchDataFromSearchableTeaserList
+                    )
+                    (Json.Decode.at [ "target", "_input", "value" ] Json.Decode.string)
+                )
             ]
             [ input [ id "search", type_ "text", placeholder (t SearchPlaceholder), css [ searchInputStyle ] ] [] ]
         ]
 
 
+searchableStringFromSearchData : Shared.SearchData -> String
```
Comment (katjam) [inline src/Theme/HeaderTemplate.elm:86]: I think this is the broken part - maybe need to init with data if it's empty.

Comment (AutumnBarlow) [reply src/Theme/HeaderTemplate.elm:86]: I have prewarned my testers that the search was being worked on so it won't be a problem for us.

Comment (katjam) [reply src/Theme/HeaderTemplate.elm:86]: Awesome. I think I just fixed it anyway. There are probably niggles to be found but it's broadly working.

**Comment by katjam** [comment]
@r-ferrier I think I've fixed it enough to merge, but would be grateful for a sanity check. If you are happy, go ahead without me. I'll try to be online. Catching a bus at 09.30.

File: src/Page/Search/Data.elm [hunk 2]
```diff
@@ -133,6 +229,32 @@ storyImagefromTeaserImage maybeImages =
...
+searchableFromStringDecoder itemType =
+    case itemType of
+        "action" ->
+            Json.Decode.succeed Shared.Action
+
+        "guide" ->
+            Json.Decode.succeed Shared.Guide
+
+        "story" ->
+            Json.Decode.succeed Shared.Guide
```
Comment (katjam) [inline src/Page/Search/Data.elm:252]:
```suggestion
            Json.Decode.succeed Shared.Story
```

**Review by r-ferrier — APPROVED** [review]
I'm merging this but it is a bit broken - searches which return 0 results show as returning all of the results. I'm writing a fix for this and will push it up in the next half an hour!

**Comment by katjam** [comment]
Great catch. Was there one more thing outstanding in the plan on #309 that
needs converting to issue? I did have some problems with the web component
causing app to freeze intermittently so am probably going to refactor into
a port voluntarily.

**Comment by r-ferrier** [comment]
The outstanding issue is that the search box is on every page.

I've been trying to get this button working but I think I'm going to have to leave it - I just can't for the life of me get a key listener to fire properly on the input - probably something to do with the search setup making use of the shadow dom, which is not something i've worked with before - I think I need someone with more front end experience to explain why! 

I also got slight stuck because we had another bug which meant we were only ever looking at the results from the last search every time we reran the search. Have fixed this so will push as a separate fix and change the text & link to 'explore the hub' for now.

**Comment by katjam** [comment]
I think you are right. I may find time to move this weekend or just leave
it be. Don't want to break right before launch.

**Pull request merged**